test/state/axisSelectors.spec.tsx

```diff
@@ -64,3 +64,249 @@ describe('selectAxisScale', () => {
     });
   });
 });
+
+describe('selectDomainOfDataByKey', () => {
+  it('should return undefined when called outside of Redux context', () => {
+    expect.assertions(1);
+    const Comp = (): null => {
+      const result = useAppSelector(state => selectDomainOfDataByKey(state, 'xAxis', '0'));
+      expect(result).toBeUndefined();
+      return null;
+    };
+    render(<Customized component={Comp} />);
+  });
+
+  it('should return undefined when called with initial state', () => {
+    const initialState: RechartsRootState = createRechartsStore().getState();
+    const result = selectDomainOfDataByKey(initialState, 'xAxis', '0');
+    expect(result).toBeUndefined();
+  });
+
+  it('should return undefined if there is no data in the chart', () => {
+    const spy = vi.fn();
+    const Comp = (): null => {
+      const result = useAppSelector(state => selectDomainOfDataByKey(state, 'xAxis', '0'));
+      spy(result);
+      return null;
+    };
+    render(
+      <BarChart data={[]} width={100} height={100}>
+        <Bar dataKey="uv" />
+        <XAxis dataKey="name" />
+        <Customized component={Comp} />
+      </BarChart>,
+    );
+    expect(spy).toHaveBeenCalledWith(undefined);
+  });
+
+  describe('XAxis with type = number', () => {
+    it('should return highest and lowest number of the chart root data based on the axis dataKey', () => {
+      const spy = vi.fn();
+      const Comp = (): null => {
+        const result = useAppSelector(state => selectDomainOfDataByKey(state, 'xAxis', '0'));
+        spy(result);
+        return null;
+      };
+      render(
+        <BarChart data={PageData} width={100} height={100}>
+          <XAxis dataKey="uv" type="number" />
+          <Customized component={Comp} />
```

Comment:
Won't these all fail later? Why can't we use renderHook?

Reply:
Yes all of these will fail. They will be trivial to replace however.